crates/matrix-sdk/src/event_cache/linked_chunk.rs

```diff
@@ -590,18 +590,17 @@ impl ChunkIdentifierGenerator {
     /// Generate the next unique identifier.
     ///
     /// Note that it can fail if there is no more unique identifier available.
-    /// In this case, `Result::Err` contains the previous unique identifier.
-    pub fn generate_next(&self) -> Result<ChunkIdentifier, ChunkIdentifier> {
+    /// In this case, this method will panic.
+    pub fn generate_next(&self) -> ChunkIdentifier {
```

Comment:
For bonus points: there's an argument that this struct is a generator, so this method could just be called `next()` :eyes:

Reply:
Renamed to `next`.